install_scripts/templates/common/kubernetes.sh

```diff
@@ -1255,12 +1255,29 @@ exportKubeconfig() {
 # Arguments:
 #   registry host
 # Returns:
-#   None
+#   ADDED_INSECURE_REGISTRY
 #######################################
+ADDED_INSECURE_REGISTRY=0
 addInsecureRegistry() {
     if grep -q "insecure-registry $1" /etc/systemd/system/docker.service.d/replicated-registry.conf 2>/dev/null ; then
         return
     fi
+    if grep "insecure-registries" /etc/docker/daemon.json 2>/dev/null | grep -q "$1" ; then
+        return
+    fi
+
+    # prefer to configure using /etc/docker/daemon.json since that does not require a docker restart
+    # but don't attempt to edit json if that file already exists.
+    if [ ! -f "/etc/docker/daemon.json" ]; then
+        cat <<EOF >/etc/docker/daemon.json
+{
+    "insecure-registries": ["$1"]
+}
+EOF
+        kill -SIGHUP $(pidof dockerd) || true
```

Comment:
why not systemctl?

Reply:
ok, I'll test this: `systemctl kill -s HUP --kill-who=main dockerd.service`